snapcraft/tests/fixture_setup.py

```diff
@@ -479,10 +481,15 @@ def call_effect(*args, **kwargs):
                             }).encode('utf-8')
                 return '[]'.encode('utf-8')
             elif args[0][:2] == ['lxc', 'list'] and self.fail_on_remote:
-                    raise CalledProcessError(returncode=255, cmd=args[0])
+                raise CalledProcessError(returncode=255, cmd=args[0])
             elif args[0][:2] == ['lxc', 'init']:
                 self.name = args[0][3]
                 self.status = 'Stopped'
+            # Fail on an actual snapcraft command and not the command
+            # for the installation of it.
+            elif ('snapcraft' in args[0] and 'apt-get' not in args[0]
+                  and self.fail_on_snapcraft_run):
+                raise CalledProcessError(returncode=255, cmd=args[0])
```

Comment:
I will have to move everything out of the fixture though, this happens late in the game.